x-pack/plugin/rank-rrf/src/main/java/org/elasticsearch/xpack/rank/rrf/RRFRankBuilder.java

```diff
@@ -55,6 +60,13 @@ public class RRFRankBuilder extends RankBuilder {
     }
 
     public static RRFRankBuilder fromXContent(XContentParser parser) throws IOException {
+        if (parser.getRestApiVersion().matches(RestApiVersion.onOrAfter(RestApiVersion.V_8))) {
+            deprecationLogger.warn(
+                DeprecationCategory.API,
+                "rank_api",
+                "Using [rank] is deprecated and will be removed in a future release"
```

Comment:
I've moved this from `SearchSourceBuilder` here, to mark just  `rank: rrf` as deprecated.  Happy to move it back if it'd be best to mark `rank` as a whole deprecated.

Reply:
ALL rank is going away. The only one was rrf right? We are removing the whole thing, only `retriever` remains

Reply:
Moved deprecation check in parsing with the following warning
```
        - "Deprecated field [rank] used, this field is unused and will be removed entirely in a future release. Please consider using the appropriate [retriever] to rank your results."

```